doc/advanced-configuration/Listener-modules.md

```diff
@@ -99,6 +99,10 @@ Unlike `ejabberd_c2s`, it doesn't use `ejabberd_receiver` or `ejabberd_listener`
 ### HTTP module: `mod_cowboy`
 
 This module provides an additional routing layer on top of HTTP(s) or WS(S) protocols.
+Regular cowboy handlers cannot cleanly handle both protocols on the same port. 
```

Comment:
I think it's perfectly fine to have a WS and HTTP handlers attached to the same listener (port) but different resources.
@michalwski,  wasn't it about the **same resource**?

Reply:
It is all possible only not in a clean way, which is sth subjective though. IMO clean cowboy handlers operates on single transport WS or HTTP (not both) and possible single or at least (similar resources). From what I understand `mod_cowboy` itself may not be clean but makes it easier and cleaner for the layers on top of it.

Reply:
I think we should rephrase it into something like this:

It allows other HTTP/WS modules to coexist under the same URL on the single port.
Packets are forwarded to them based on protocol.
This mechanism is transparent for actual handlers so the path sharing does not require additional code.
Example: If you wish, you can use BOSH and WS XMPP handlers (`mod_bosh`, `mod_websockets`) on single port and URL without any code modifications.

What do you think?

Reply:
Yes, @fenek described it in the most accurate way in my opinion.

Reply:
It allows other HTTP/WS modules to coexist under the same URL on the single port.
Packets are forwarded to them based on the protocol.
This mechanism is transparent to actual handlers so the path sharing does not require any additional code.
Example: If you wish, you can use BOSH and WS XMPP handlers (mod_bosh, mod_websockets) on a single port and a URL without any code modifications.